hazelcast/src/main/java/com/hazelcast/executor/impl/operations/AbstractCallableTaskOperation.java

```diff
@@ -58,32 +50,25 @@ public AbstractCallableTaskOperation(String name, String uuid, Data callableData
     }
 
     @Override
-    public final void beforeRun() throws Exception {
-        returnsResponse = false;
+    public final CallStatus call() {
+        Callable callable = loadCallable();
+        DistributedExecutorService service = getService();
+        service.execute(name, uuid, callable, this);
+        return OFFLOADED;
+    }
 
-        callable = getCallable();
+    private Callable loadCallable() {
         ManagedContext managedContext = getManagedContext();
 
+        Callable callable = getNodeEngine().toObject(callableData);
         if (callable instanceof RunnableAdapter) {
             RunnableAdapter adapter = (RunnableAdapter) callable;
             Runnable runnable = (Runnable) managedContext.initialize(adapter.getRunnable());
             adapter.setRunnable(runnable);
         } else {
             callable = (Callable) managedContext.initialize(callable);
         }
-    }
-
-    /**
-     * since this operation handles responses in an async way, we need to handle serialization exceptions too
-     * @return
-     */
-    private Callable getCallable() {
-        try {
```

Comment:
If an exception is thrown while deserializing, it will be caught by the OperationRunner and no matter the 'hasResponse' result, a response is going to be send. This is changed in 3.9, so we don't need to trap the exception in this operation to get is shipped.